### PR TITLE
[BUGFIX] Icons implemented correctly

### PR DESCRIPTION
**Pull request by Madii**
Now the icons are shown the TYPO3 way

Fix: #131

### CODE REVIEW

**Comment by NamelessCoder** [comment]
Hi @Madii - thanks for the patches! Although it's a bit of a read I think you can benefit a lot from reading and using the tips in our contribution guide:

https://fluidtypo3.org/documentation/contributing/contribution-guide.html

It walks you through **all** the required steps for setting up your local working copy and for validating all your hard work before it gets pushed and describes the workflow for so-called `amending` which we do instead of creating new commits. It's a bit of extra security that gives you much quicker feedback about for example CGL issues - if you do all the steps, you'll basically always be creating formally valid pull requests on the first try from then onward.

This leaves you as well as us free to focus on the logic of the code rather than the form - where our energy is best spent ;)

PS: I noticed this one in another code change as well - in day to day communication most of us completely ignore it but once committed to code and code comments, the only correct spelling of TYPO3 is `TYPO3`; all caps. It's a trademark thing ;)

**Comment by Madii** [comment]
Ah ok...Thanks for the feedback Claus :-) ...i am not very used to CodeSniffer yet.

**Comment by Madii** [comment]
And although reading the error i don't understand this one....only 1 left and it's annoying me.

**Comment by Madii** [comment]
i ran all tests...everythin ok. still here i got this travis ci error. ;-)

File: Classes/UserFunction/ProviderField.php [hunk 1]
```diff
@@ -99,20 +100,33 @@ protected function renderOptions(array $variants) {
 	 * @return string
 	 */
 	protected function renderSelectField($parameters, $options, $selectedValue) {
-		$hasSelectedValue = (TRUE === empty($selectedValue) || TRUE === array_key_exists($selectedValue, $options));
+		$optionsIcons = array();
+		$optionsLabels = array();
+		$selectedIcon = '<img alt="" src="" /> ';
+		foreach ($options as $extensionKey => $optionsSetup){
```
Comment (NamelessCoder) [inline Classes/UserFunction/ProviderField.php]: `foreach ($options as $extensionKey => $optionsSetup) {`

**Comment by NamelessCoder** [comment]
Maybe it's not pushed? Local validation is only the first step - once that's good, you need to "squash" the commits so there's only one left. Do `git reset HEAD^1` until `git log -n1` shows the right commit, then see the guide for how to "amend" that commit and force push the result. Travis will then re-test everything.

**Comment by Madii** [comment]
well i pushed it here by editing here. normally this works. i can see the errors here.
 106 | ERROR | [x] Expected 1 space after closing parenthesis; found "" 
Don't know what he wants...that's why comitted two times more

File: Classes/UserFunction/ProviderField.php (hunk 1 above)
Comment (Madii) [inline Classes/UserFunction/ProviderField.php]: ahhh now i get it

**Comment by Madii** [comment]
Now i am banging my head at the table ....

**Comment by Madii** [comment]
Why did coverage fail?

**Comment by Madii** [comment]
ok i wait for your explanation Claus because i don't understand this one ;)

**Comment by cedricziel** [comment]
Hi @Madii you need to squash your commits together like metioned before.

You have 6 commits by now, which means you will need to rebase interactively and then choose squash for every commit but the first.

1.) Switch to your console/terminal
2.) ``git rebase -i HEAD~6`` <- an editor will come up
3.) make sure those are all your commits and ``[BUGFIX] Icons implemented correctly`` is the first one
4.) Replace the prefix ``pick`` on all but the first commit with ``s`` or ``squash``, save and close, your will be prompted for a new commit message which has all 6 commit message in it. Save & close
5.) Perform ``git push -f $origin development`` where ``$origin`` is the name of your github remote. If unsure, look it up via ``git remote -v``

Thank you for your contribution!

**Comment by Madii** [comment]
Thank you @cedricziel :)

**Comment by NamelessCoder** [comment]
Explanation for coverage drop is in https://coveralls.io/builds/2171611/source?filename=UserFunction%2FProviderField.php (you can reach it by clicking the coverage percentage icon). The entire condition that turns the `$icon` variable into an empty string if it is NULL, can simply be removed. If you're really compulsive about types, cast `(string) $icon` where the variable gets used - but it's not really necessary.

**Comment by Madii** [comment]
Ah now it works...in the night when i clicked it, it just said that it couldn't load the file from github. So i couldn't see whats wrong. Thank you @NamelessCoder

**Comment by NamelessCoder** [comment]
> in the night when i clicked it, it just said that it couldn't load the file from github.

Yep, missing config in Coveralls that I added just seconds ago ;)

The other way to get this info before Coveralls is to run `./vendor/bin/phpunit --coverage-text` (the output gets a little big but it should immediately show which classes have dropped).

**Comment by Madii** [comment]
You all happy now? :)

**Comment by NamelessCoder** [comment]
> You all happy now? :)

Very - but don't you also agree that this looks so much better and cleaner now? ;)

Good call about the IDE integration (I also do that). It could be the phpunit version... if all else fails, you could configure that project to use the phpunit from any of the ft3 extensions; it's always the same one (coming from `fluidtypo3-development` as composer dependency).

**Comment by Madii** [comment]
> Very - but don't you also agree that this looks so much better and cleaner now? ;)
Yeah i agree, although it took me a lot of time. But after all i see that this is much better ;)

**Comment by cedricziel** [comment]
you can look at fluidtypo3/development for the version of phpunit we use.

**Comment by Madii** [comment]
i already looked its the same...so i don't know where the problem lies. Seems fldtypo3/development is using some additions/extensions more. dunno.

**Comment by NamelessCoder** [comment]
Everything stated in `require-dev` of the composer manifest of fluidtypo3 packages must be available when testing. Using the phpunit plus the bootstrap script we ship automatically does that.

Note that `require-dev` is not pulled if you include the fluidtypo3 extension as a dependency. You really do need to run `composer install` from within the extension directory regardless how counter-intuitive that may seem - it is the only way to ensure that all these tests run 100% decoupled from any sort of TYPO3 context you might be testing it in; so no tests depend on your site's configuration or database contents.

**Comment by Madii** [comment]
Well i already made my peace with it that i have to use the composer phpunit :) . But thx anyway for the explenation :)

**Comment by NamelessCoder** [comment]
The good thing about using the composer dependencies we ship, is we already did all this dependency work for you ;)

Hope you'll use the new setup to create even more contributions - thx for all the ones so far! ^^

**Comment by Madii** [comment]
Yeah you all doing great ;) ...don't encourage me to spend so much time for you :P and you are welcome :)

**Pull request merged**